mucommander-core/src/main/java/com/mucommander/ui/dialog/QuestionDialog.java

```diff
@@ -38,23 +40,26 @@
 
 
 /**
- * 
+ * A generic dialog box issued to users if muCommander is in doubt how to continue i.e.
+ * dialog boxes that ask questions or require confirmations.
  *
  * @author Maxence Bernard
  */
-public class QuestionDialog extends FocusDialog implements ActionListener, DialogResult {
-	
+public class QuestionDialog extends FocusDialog implements DialogResult {
+
+    /** This value is returned by {@link #getActionValue()} when the dialog has been disposed without the user
+     * selecting a custom action */
+    public static final DialogAction DIALOG_DISPOSED_ACTION = null;
+
     /** Dialog owner */
-    private JButton buttons[];
-    private int actionValues[];
-	
-    private int retValue = DIALOG_DISPOSED_ACTION;
+    private List<JButton> buttons = new ArrayList<>();
+    private List<DialogAction> actionChoices = new ArrayList<>();
```

Comment:
Since the initialization (`init` method) is kinda optional (not enforced by c-tor), I don't like to keep such members uninitialized/null as it later on it would be nice check against null each time we access it.

Reply:
going to make them final and initialized. I can change if you insist tho :)